lib/credo/check/readability/cond_final_condition.ex

```
@@ -0,0 +1,86 @@
+            x > y -> 0
+            true -> 1
+          end
+      """,
+      params: [
+        final_condition_value: "Set the expected value for the final condition"
+      ]
+    ]
+
+  @doc false
+  @impl true
+  def run(%SourceFile{} = source_file, params) do
+    issue_meta = IssueMeta.for(source_file, params)
+
+    final_condition_value = Params.get(params, :final_condition_value, __MODULE__)
+
+    Credo.Code.prewalk(source_file, &traverse(&1, &2, issue_meta, final_condition_value))
+  end
+
+  defp traverse({:cond, meta, arguments} = ast, issues, issue_meta, final_condition_value) do
+    conditions =
+      arguments
+      |> Credo.Code.Block.do_block_for!()
+      |> List.wrap()
+
+    if conditions
+      |> List.last()
+      |> catchall_other_than_value?(final_condition_value) do
+      {ast, issues ++ [issue_for(issue_meta, meta[:line], :cond)]}
+    else
+      {ast, issues}
+    end
+  end
+
+  defp traverse(ast, issues, _issue_meta, _min_pipeline_length) do
+    {ast, issues}
+  end
+
+  defp catchall_other_than_value?({:->, _meta, [[value], _args]}, value), do: false
+  # Integer literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, ['{', _args]}, _value), do: true
+  # Binary literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, [[binary], _args]}, _value) when is_binary(binary), do: true
+  # List literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, [[list], _args]}, _value) when is_list(list), do: true
+  # Map literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, [[{:%{}, _meta2, []}], _args]}, _value), do: true
+  # Tuple literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, [[{:{}, _meta2, _values}], _args]}, _value), do: true
+  # Atom literal catch-all clause
+  defp catchall_other_than_value?({:->, _meta, [[name], _args]}, _value) when is_atom(name), do: true
+  defp catchall_other_than_value?(_clause, _value), do: false
```

If the clauses "above" this one are "catch alls" for all cases, what are examples where we need this clause?

---

This is for a non catch-all clause. A contrived example:

```
 cond do
   n > 5 ->
     "greater than 5"
   n >= 0 ->
     "positive"
   n < 0 ->
     "negative"
end
```

The last clause is not a catchall and is not composed of a single literal. I will add a comment for this final clause. Let me know if you think I should rename this function.